tests/test_check_parallel.py

```diff
@@ -401,3 +477,69 @@ def test_compare_workers_to_single_proc(self, num_files, num_jobs, num_checkers)
         assert (
             stats_check_parallel == expected_stats
         ), "The lint is returning unexpected results, has something changed?"
+
+    @pytest.mark.parametrize(
+        "num_files,num_jobs,num_checkers",
+        [
+            (2, 2, 1),
+            (2, 2, 2),
+            (2, 2, 3),
+            (3, 2, 1),
+            (3, 2, 2),
+            (3, 2, 3),
+            (3, 1, 1),
+            (3, 1, 2),
+            (3, 1, 3),
+            (3, 5, 1),
+            (3, 5, 2),
+            (3, 5, 3),
+            (10, 2, 1),
+            (10, 2, 2),
+            (10, 2, 3),
+            (2, 10, 1),
+            (2, 10, 2),
+            (2, 10, 3),
+        ],
+    )
+    def test_map_reduce(self, num_files, num_jobs, num_checkers):
```

Comment:
You could probably, and more neatly, either:

1. merge this with my original (awful?) code (put `linter.register_checker(ExtraParallelTestChecker(linter))` next to `linter.register_checker(ExtraSequentialTestChecker(linter))`
2. create a shared function that accepts the `ExtraSequentialTestChecker`/`ExtraParallelTestChecker` types as params?

Either way, I apologize for writing the code this test is based on, it was the only way I could think to do it in the time I had.

Reply:
Merging both tests, that does almost the same thing, would take less lines, but more complicated.

The first one can pass while the second can fail (before this PR), if we merge them I fear it'll add complexity while debugging the day it'll fail again.

Reply:
Would something like this be more readable:

```python
file_infos = _gen_file_datas(num_files)

checkers = [
    ParallelTestChecker,
    ExtraParallelTestChecker,
    ThirdParallelTestChecker,
][:num_checkers]

# Establish the baseline:
linter = PyLinter(reporter=Reporter())
for checker in checkers[:num_checkers]:
    linter.register_checker(checker(linter))
assert linter.config.jobs == 1, "jobs>1 are ignored when calling _check_files"
linter._check_files(linter.get_ast, file_infos)
stats_single_proc = linter.stats

# Run the same in parallel:
linter = PyLinter(reporter=Reporter())
for checker in checkers[:num_checkers]:
    linter.register_checker(checker(linter))
check_parallel(linter, jobs=num_jobs, files=file_infos, arguments=None)
stats_check_parallel = linter.stats

assert (
    stats_single_proc["by_msg"] == stats_check_parallel["by_msg"]
), "Single-proc and check_parallel() should return the same thing"

```

? It's 9 lines shorter, mostly due to the reduced indentation that allow to use longer lines.